web-common/src/features/scheduled-reports/ScheduledReportDialog.svelte

```diff
@@ -115,31 +125,49 @@
   const schema = yup(
     object({
       title: string().required("Required"),
+      webOpenMode: string().required("Required"),
       emailRecipients: array().of(string().email("Invalid email")),
       enableSlackNotification: boolean(), // Needed to get the type for validation
       slackChannels: array().of(string()),
       slackUsers: array().of(string().email("Invalid email")),
       columns: array().of(string()).min(1),
-    }).test(
-      "at-least-one-recipient",
-      "At least one email recipient, slack user, or slack channel is required",
-      function (value) {
-        // Check if at least one array has non-empty values
-        const hasEmailRecipients = value.emailRecipients
-          ? value.emailRecipients.filter(Boolean).length > 0
-          : false;
-        if (!value.enableSlackNotification) return hasEmailRecipients;
+    })
+      .test(
+        "at-least-one-recipient",
+        "At least one email recipient, slack user, or slack channel is required",
+        function (value) {
+          // Check if at least one array has non-empty values
+          const hasEmailRecipients = value.emailRecipients
+            ? value.emailRecipients.filter(Boolean).length > 0
+            : false;
+          if (!value.enableSlackNotification) return hasEmailRecipients;
 
-        const hasSlackUsers = value.slackUsers
-          ? value.slackUsers.filter(Boolean).length > 0
-          : false;
-        const hasSlackChannels = value.slackChannels
-          ? value.slackChannels.filter(Boolean).length > 0
-          : false;
+          const hasSlackUsers = value.slackUsers
+            ? value.slackUsers.filter(Boolean).length > 0
+            : false;
+          const hasSlackChannels = value.slackChannels
+            ? value.slackChannels.filter(Boolean).length > 0
+            : false;
 
-        return hasEmailRecipients || hasSlackUsers || hasSlackChannels;
-      },
-    ),
+          return hasEmailRecipients || hasSlackUsers || hasSlackChannels;
+        },
+      )
+      .test(
+        "as-recipients-in-project",
+        "Recipients should be part of the project when running as recipient",
```

Comment:
"should be" makes it sound optional. How about "must be" if this is a required condition for form submission?